src/main/scala/firrtl/transforms/PropagatePresetAnnotations.scala

```diff
@@ -0,0 +1,448 @@
+// See LICENSE for license details.
+
+package firrtl
+package transforms
+
+import firrtl.{Utils}
+
+
+import firrtl.PrimOps._
+import firrtl.ir._
+import firrtl.ir.{AsyncResetType}
+import firrtl.annotations._
+import firrtl.options.{Dependency, PreservesAll}
+
+import scala.collection.mutable
+
+object PropagatePresetAnnotations {
+  val advice = "Please Note that a Preset-annotated AsyncReset shall NOT be casted to other types with any of the following functions: asInterval, asUInt, asSInt, asClock, asFixedPoint, asAsyncReset."
+  case class TreeCleanUpOrphanException(message: String) extends FirrtlUserException(s"Node left an orphan during tree cleanup: $message $advice")
+}
+
+/** Propagate PresetAnnotations to all children of targeted AsyncResets 
+  * Leaf Registers are annotated with PresetRegAnnotation
+  * All wires, nodes and connectors along the way are suppressed
+  *
+  * Processing of multiples targets are NOT isolated from one another as the expected outcome does not differ
+  * Annotations of leaf registers, wires, nodes & connectors does indeed not depend on the initial AsyncReset reference 
+  * The set of created annotation based on multiple initial AsyncReset PresetAnnotation 
+  *
+  * This transform consists of 2 successive walk of the AST
+  * I./ Propagate 
+  *    - 1./ Create all AsyncResetTrees 
+  *    - 2./ Leverage them to annotate register for specialized emission & PresetTree for cleanUp
+  * II./ CleanUpTree
+  *    - clean up all the intermediate nodes (replaced with EmptyStmt)
+  *    - raise Error on orphans (typically cast of Annotated Reset)
+  *    - disconnect Registers from their reset nodes (replaced with UInt(0))
+  *
+  * Thanks to the clean-up phase, this transform does not rely on DCE
+  *
+  * @note This pass must run before InlineCastsTransform
+  */
+class PropagatePresetAnnotations extends Transform with PreservesAll[Transform] {
+  def inputForm = UnknownForm
+  def outputForm = UnknownForm
+  
+  override val prerequisites = firrtl.stage.Forms.LowFormMinimumOptimized ++
+    Seq( Dependency[BlackBoxSourceHelper],
+         Dependency[FixAddingNegativeLiterals],
+         Dependency[ReplaceTruncatingArithmetic])
+
+  override val optionalPrerequisites = firrtl.stage.Forms.LowFormOptimized
+
+  override val dependents = Seq.empty
+  
+  
+  import PropagatePresetAnnotations._
+  
+  type TargetSet = mutable.HashSet[ReferenceTarget]
+  type TargetMap = mutable.HashMap[ReferenceTarget,String]
+  type TargetSetMap = mutable.HashMap[ReferenceTarget, TargetSet]
+  
+  val toCleanUp = new TargetSet()
```

Comment:
My bad for not noticing this sooner, it would be nice for everything in here to be private just so that we can easily change them without violating binary compatibility in patch releases.

I don't want to subject you to yet another 24-hour turn around time though, so I'm going to do it then approve and merge 🙂